### PR TITLE
Document module function parseTokenizer 

### PR DESCRIPTION
**Pull request by Borewit**
* As requested [here](https://github.com/sindresorhus/file-type/pull/261#discussion_r360573288), documented `FileType.parseTokenizer()`.
* Updated API (method and parameters) documentation

### CODE REVIEW

File: core.d.ts [hunk 1]
```diff
@@ -266,10 +267,19 @@ declare namespace core {
   The file type is detected by checking the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files) of the buffer.
 
 	@param stream - Node.js readable stream
-	@returns The detected file type and MIME type or `undefined` when there was no match.
+	@returns The detected file type and MIME type via a promise, or `undefined` when there was no match.
 	 */
 	function fromStream(stream: Readable): Promise<core.FileTypeResult | undefined>;
 
+	/**
+	 * Parse audio from a ITokenizer source.
```
Comment (sindresorhus) [inline core.d.ts]: Audio only? Why?

File: core.d.ts [hunk 2]
```diff
@@ -257,7 +258,7 @@ declare namespace core {
  	If file access is available, it is recommended to use `fromFile()` instead
 
 	@param buffer - It works best if the buffer contains the entire file, it may work with a smaller portion as well
-	@returns The detected file type and MIME type or `undefined` when there was no match.
+	@returns The detected file type and MIME type via a promise, or `undefined` when there was no match.
```
Comment (sindresorhus) [inline core.d.ts]: > via a promise

This is moot as the return type already says it's a promise.

File: readme.md [hunk 3]
```diff
@@ -92,6 +92,33 @@ const fileType = require('file-type');
 })();
 ```
 
+Or from a _tokenizer_:
+
+A [_tokenizer_](https://github.com/Borewit/strtok3#tokenizer) propagates the internal read functions, 
+allowing alternative transport mechanisms, to access files, to be implemented and used.
+
+An example is [@tokenizer/http](https://github.com/Borewit/tokenizer-http) which requests data using range-requests.
+A difference with a conventional stream, is that it for example allows to only read the first 6, and the last 128 bytes.
```
Comment (sindresorhus) [inline readme.md]: The docs in the TS file and readme should be in sync.

File: readme.md [hunk 4]
```diff
@@ -92,6 +92,33 @@ const fileType = require('file-type');
...
+
+const audioTrackUrl = 'https://test-audio.netlify.com/Various%20Artists%20-%202009%20-%20netBloc%20Vol%2024_%20tiuqottigeloot%20%5BMP3-V2%5D/01%20-%20Diablo%20Swing%20Orchestra%20-%20Heroines.mp3';
+
+(async () => {
+  const httpTokenizer = HttpTokenizer.fromUrl(audioTrackUrl, config);
+  await httpTokenizer.init();
+
+  const fileType = await FileType.fromTokenizer(httpTokenizer);
+  console.log(fileType);
+  // { ext: 'mp3', mime: 'audio/mpeg' }
```
Comment (sindresorhus) [inline readme.md]: Use tab-indentation.

Comment (Borewit) [reply readme.md]: Oh yes, I cannot find an option in IntelliJ to override tab indents for markdown.

File: core.d.ts (hunk 1 above)
Comment (Borewit) [reply core.d.ts]: Copy past mistake

File: readme.md (hunk 3 above)
Comment (Borewit) [reply readme.md]: I missed the API documentation. I updated that path with the Typescript definitions.

File: core.d.ts (hunk 2 above)
Comment (Borewit) [reply core.d.ts]: But in the readme I kept the 

> _Returns a `Promise` resolving the _...

as the promise is not obvious

File: readme.md (hunk 4 above)
Comment (Borewit) [reply readme.md]: Fixed them one more time

File: readme.md [hunk 5]
```diff
@@ -123,44 +154,83 @@ const url = 'https://upload.wikimedia.org/wikipedia/en/a/a9/Example.jpg';
...
+
+If file access is available, it is recommended to use `fromFile()` instead
+
+| Parameter | Type                                                                                                                                                                                                                                                                            | Description                       |
+|-----------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------|
+| path      | [`Buffer`](https://nodejs.org/api/buffer.html) or [`Uint8Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array)  or [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) | Buffer representing file content  |
```
Comment (sindresorhus) [inline readme.md]: I don't want to use Markdown tables. They're annoying to maintain.

File: readme.md [hunk 6]
```diff
@@ -90,6 +90,37 @@ const FileType = require('file-type');
 })();
 ```
 
+Or from a _tokenizer_:
```
Comment (sindresorhus) [inline readme.md]: I think this should be moved to under the `.fromTokenizer()` docs as it's not something most would need.

File: readme.md [hunk 7]
```diff
@@ -90,6 +90,37 @@ const FileType = require('file-type');
...
+
+A [_tokenizer_](https://github.com/Borewit/strtok3#tokenizer) propagates the internal read functions, 
+allowing alternative transport mechanisms, to access files, to be implemented and used.
+
+An example is [@tokenizer/http](https://github.com/Borewit/tokenizer-http),
+which requests data using [HTTP-range-requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests).
+A difference with a conventional stream and the [_tokenizer_](https://github.com/Borewit/strtok3#tokenizer),
+is that it is able to _ignore_ (seek, fast-forward) in the stream.
```
Comment (sindresorhus) [inline readme.md]: Don't hard-wrap.

File: readme.md [hunk 8]
```diff
@@ -123,44 +154,83 @@ const url = 'https://upload.wikimedia.org/wikipedia/en/a/a9/Example.jpg';
...
 ## API
 
-### fileType(input)
 
-Returns an `object` with:
+### FileType.fromBuffer(buffer)
+Detect the file type of a `Buffer`/`Uint8Array`/`ArrayBuffer`.
+The file type is detected by checking the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files) of the buffer.
+
+If file access is available, it is recommended to use `fromFile()` instead
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
If file access is available, it is recommended to use `fromFile()` instead.
```

File: readme.md (hunk 5 above)
Comment (Borewit) [reply readme.md]: Converted to nested list

File: readme.md (hunk 6 above)
Comment (Borewit) [reply readme.md]: Done

**Pull request merged**